src/solidlsp/language_servers/common.py

```diff
@@ -63,39 +63,57 @@ def install(self, logger: LanguageServerLogger, target_dir: str) -> dict[str, st
             if dep.url:
                 self._install_from_url(dep, logger, target_dir)
             if dep.command:
-                self._run_command(dep.command, target_dir)
+                self._run_command(dep.command, logger, target_dir)
             if dep.binary_name:
                 results[dep.id] = os.path.join(target_dir, dep.binary_name)
             else:
                 results[dep.id] = target_dir
         return results
 
     @staticmethod
-    def _run_command(command: str, cwd: str) -> None:
+    def _run_command(command: str | list[str], logger: LanguageServerLogger, cwd: str) -> None:
         kwargs = {}
         if PlatformUtils.get_platform_id().is_windows():
             kwargs["creationflags"] = subprocess.CREATE_NO_WINDOW  # type: ignore
         else:
             import pwd
 
             kwargs["user"] = pwd.getpwuid(os.getuid()).pw_name
-        log.info("Running command '%s' in '%s'", command, cwd)
-        completed_process = subprocess.run(
-            command,
-            shell=True,
-            check=True,
-            cwd=cwd,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.STDOUT,
-            **kwargs,
-        )
-        if completed_process.returncode != 0:
-            log.warning("Command '%s' failed with return code %d", command, completed_process.returncode)
-            log.warning("Command output:\n%s", completed_process.stdout)
+
+        if isinstance(command, list):
+            command_parts = command
```

Comment:
let's keep simple interfaces. If you need a list for solving this issue, the interface should be changed to `command: list[str]` and the callers should be responsible for creating the list.

Reply:
done